lib/tokenizer/rainscriptTokenizerRules.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * The tokenizer rules, ordered by priority. All tokenizer rules go here.
+ *
+ * @type {{pattern: RegExp, type: any}[]}
+ * @property {RegExp} pattern a regular expression to be matched by a rule
+ * @property {*}      type    the type of the next token that will be returned
+ *                            when the pattern matches correctly
+ */
+export const rainscriptTokenizerRules = [
+  // quoted string literals
+  { pattern: /'(\\.|[^\\'])*'/, type: 'STR_1' },
```

Comment:
why not use an object and use either
```js
'STR_1': /'(\\.|[^\\'])*'/,
```
or
```js
[/'(\\.|[^\\'])*'/.source]: 'STR_1',
```

Reply:
An array of objects is preferred since **order is important** for tokenizer rules (first rule first served). Also, this is more flexible for advanced lexer arguments, for error messages for example.

Example:
```js
[
  // first check if valid single-quoted string
  { pattern: /'(\\.|[^\\'])*'/, type: 'STR_1' },
  
  // fallback for unfinished single-quote, tokenize as an error immediately
  { pattern: /'(\\.|[^\\'])*/, type: 'SUPPOSED_TO_BE_STR_1', error: 'unfinished quote' } 

  // ...
]
```

Reply:
Ugh regex is so unreadable. Maybe you can use a regex builder [like this](https://www.npmjs.com/package/regularity)?